src/pybind/mgr/cephadm/module.py

```diff
@@ -1184,6 +1188,15 @@ def get_unique_name(self, daemon_type, host, existing, prefix=None,
                 continue
             return name
 
+    def get_service_name(self, daemon_type, daemon_id, host):
+        # type: (str, str, str) -> (str)
+        """
+        Returns the generic service name
+        """
+        p = re.compile(r'(.*)\.%s.*' % (host))
+        p.sub(r'\1', daemon_id)
+        return '%s.%s' % (daemon_type, p.sub(r'\1', daemon_id))
```

Comment:
We have a similar function already there:

https://github.com/ceph/ceph/blob/7f0c3c4f37ca50efed57292e332426308aded05b/src/python-common/ceph/deployment/service_spec.py#L353-L357

Reply:
well except `service_id` does not contain the `host` str whereas `daemon_id` does ...

this awkward piece of code: https://github.com/ceph/ceph/pull/33581#discussion_r392729391